Added detail about updateRow return content (#19554)
mui/mui-x: docs/data/data-grid/server-side-data/index.md

The Data Source supports an optional `updateRow()` method for updating data on the server.

> This method returns a promise that resolves when the row is updated.
> The promise must contain a row. If the update is successful it shall be the updated row, else the previous row.

Reviewer: ```suggestion
This method returns a promise that resolves with the updated row or rejects on update error.
```
